packages/angular/ssr/test/app-engine_spec.ts

```diff
@@ -0,0 +1,136 @@
+import '@angular/compiler';
+/* eslint-enable import/no-unassigned-import */
+
+import { AngularServerApp } from '../src/app';
+import { AngularAppEngine } from '../src/app-engine';
+import { setAngularAppEngineManifest } from '../src/manifest';
+import { setAngularAppTestingManifest } from './testing-utils';
+
+describe('AngularAppEngine', () => {
+  let appEngine: AngularAppEngine;
+  // Enable dev mode for testing purposes.
+  // @ts-expect-error Property 'isDevMode' does not exist on type 'typeof AngularAppEngine'.
+  AngularAppEngine.isDevMode = true;
+
+  describe('Localized app', () => {
+    const LOCALES = ['it', 'en'];
+
+    beforeAll(() => {
+      setAngularAppEngineManifest({
+        entryPoints: new Map(
+          LOCALES.map((locale) => [
+            locale,
+            async () => {
+              setAngularAppTestingManifest(locale);
+
+              return { AngularServerApp };
+            },
+          ]),
+        ),
+        basePath: '',
+      });
+
+      appEngine = new AngularAppEngine();
+    });
+
+    it('should return null for requests to unknown pages', async () => {
+      const request = new Request('https://example.com/unknown/page');
+      const response = await appEngine.render(request);
+      expect(response).toBeNull();
+    });
+
+    it('should return null for requests with unknown locales', async () => {
+      const request = new Request('https://example.com/es/home');
+      const response = await appEngine.render(request);
+      expect(response).toBeNull();
+    });
+
+    for (const locale of LOCALES) {
+      const upperCaseLocale = locale.toUpperCase();
+
+      describe(`when requested locale is ${upperCaseLocale}`, () => {
+        it(`should return a rendered page`, async () => {
+          const request = new Request(`https://example.com/${locale}/home`);
+          const response = await appEngine.render(request);
+          expect(await response?.text()).toContain(`Home works ${upperCaseLocale}`);
+        });
+
+        it(`should correctly render the content when the URL ends with 'index.html'`, async () => {
+          const request = new Request(`https://example.com/${locale}/home/index.html`);
+          const response = await appEngine.render(request);
+          expect(await response?.text()).toContain(`Home works ${upperCaseLocale}`);
+        });
+
+        // TODO: (Angular will render this as it will render all routes even unknown routes)
+        // ERROR RuntimeError: NG04002: Cannot match any routes. URL Segment: 'unknown/page'
+        xit(`should return null for requests to unknown pages`, async () => {
+          const request = new Request(`https://example.com/${locale}/unknown/page`);
+          const response = await appEngine.render(request);
+          expect(response).toBeNull();
+        });
+
+        it('should return null for requests to file-like resources', async () => {
+          const request = new Request(`https://example.com/${locale}logo.png`);
```

Comment:
Nit: I think this was supposed to be `https://example.com/${locale}/logo.png` (note the extra `/`).